files/usr/share/ucode/aredn/configuration.uc

```diff
@@ -428,3 +428,48 @@ export function unescapeString(s)
     }
     return s;
 };
+
+const backupFilename = "/tmp/backup.tar.gz";
+
+export function backup()
+{
+    const fi = fs.open("/etc/arednsysupgrade.conf");
+    if (!fi) {
+        return null;
+    }
+    const fo = fs.open("/tmp/sysupgradefilelist", "w");
+    if (!fo) {
+        fi.close();
+        return null;
+    }
+    for (let l = fi.read("line"); length(l); l = fi.read("line")) {
+        if (!match(l, "^#") && !match(l, "^/etc/config/") && fs.access(trim(l))) {
```

Comment:
More for my own edification, so you're welcome to ignore this but..

This implies that `/etc/arednsysupgrade.conf` contains a list of relevant config files for the node. It also implies that only some of the lines in that file refer to files in `/etc/config/`, and that only those files should be kept for backup purposes.

Why the extra steps here? Would it be okay to just compress the entirety of the `/etc/config` directory and send it down?

Reply:
AREDN doesnt use /etc/config like OpenWRT does. We dont backup any of /etc/config right now (if you look at the current release you'll see that's not a thing). However, we will start to include some very select bits of it in order to reduce the firmware upgrade time. But we dont need this and we dont want it in a backup, so it's excluded here.